### ISSUE TITLE
Runtime error with Firebase-UI 4.3.1/4.3.1

### ISSUE THREAD

**Issue by noxtrip** [firebase/FirebaseUI-Android#1600]
### Step 2: Describe your environment

  * Android device: 
  * Android OS version: _____
  * Google Play Services version: 4.0.1
  * Firebase/Play Services SDK version: core 16.0.7
  * FirebaseUI version: 4.3.0/4.3.1
  
### Step 3: Describe the problem:

Hello. I googled and read probably everything about similiar issues but did not find anything. My 
app crashes when trying to use email logon while facebook and google login methods are working.

#### Observed Results:

When I click on "Login with email" option in generated UI of Firebase-UI module app just crashes with following errors.

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{org.godotengine.bacterio/com.firebase.ui.auth.ui.email.EmailActivity}: android.view.InflateException: Binary XML file line #34: Binary XML file line #34: Error inflating class android.support.design.widget.TextInputLayout
Caused by: android.view.InflateException: Binary XML file line #34: Binary XML file line #34: Error inflating class android.support.design.widget.TextInputLayout
Caused by: android.view.InflateException: Binary XML file line #34: Error inflating class android.support.design.widget.TextInputLayout
Caused by: android.content.res.Resources$NotFoundException: Drawable org.godotengine.godot:drawable/design_password_eye with resource ID #0x7f07008f
Caused by: android.content.res.Resources$NotFoundException: File res/drawable-v21/design_password_eye.xml from drawable resource ID #0x7f07008f
``` 

#### Relevant Code:

My project build.gradle
[build.gradle.txt](https://github.com/firebase/FirebaseUI-Android/files/2968457/build.gradle.txt)

My project manifest
[AndroidManifest.xml.txt](https://github.com/firebase/FirebaseUI-Android/files/2968459/AndroidManifest.xml.txt)

I have checked the APK and I can see resources mentioned in error in RES folder.
![x](https://user-images.githubusercontent.com/48566896/54391284-a8146300-46a4-11e9-9884-b04d9b66f401.png)

Any help/advice would be appricieted.

**Comment by samtstern** [comment]
AS you noticed these lines are very suspicious:
```
Caused by: android.content.res.Resources$NotFoundException: Drawable org.godotengine.godot:drawable/design_password_eye with resource ID #0x7f07008f
Caused by: android.content.res.Resources$NotFoundException: File res/drawable-v21/design_password_eye.xml from drawable resource ID #0x7f07008f
```

What is `design_password_eye`, any idea where it comes from?  I don't think it's part of FirebaseUI or used in FirebaseUI.

Also are you passing a custom theme to FirebaseUI?  If so what is the specification?

**Comment by noxtrip** [comment]
> AS you noticed these lines are very suspicious:
> 
> ```
> Caused by: android.content.res.Resources$NotFoundException: Drawable org.godotengine.godot:drawable/design_password_eye with resource ID #0x7f07008f
> Caused by: android.content.res.Resources$NotFoundException: File res/drawable-v21/design_password_eye.xml from drawable resource ID #0x7f07008f
> ```
> What is `design_password_eye`, any idea where it comes from? I don't think it's part of FirebaseUI or used in FirebaseUI.
> 
> Also are you passing a custom theme to FirebaseUI? If so what is the specification?

Hi Sam, first thank you very much for quick replay.

I think lines are coming from FirebaseUI or at least indirectly.  From what I have found `design_password_eye or design_password_eye.xml` are drawable resources of com.android.support:design:28.0.0 that is used by FirebaseUI. 
I have found this issue [https://stackoverflow.com/questions/42789175/textinputlayout-crashing-in-android-lollipop-and-above/47342372](url) but I am using newer support library and newer gradle version then mentioned there.

For custom theme I tried to call AuthUI.getInstance() with and also without  part `.setTheme(R.style.LoginTheme)` while my style is simple:
`<style name="LoginTheme" parent="FirebaseUI">
    <item name="windowNoTitle">true</item>
</style>`

I am not sure if I could somhow change FirebaseUI layout so I could try answer from link I posted higher:
`So, we can avoid Resources$NotFoundException by setting app:passwordToggleDrawable="@null"`

Thank you very much.

**Comment by samtstern** [comment]
Ok I want to determine if this is actually an issue with FirebaseUI or just an issue with your app that is becoming visible through FirebaseUI.  Could you try adding some views like this to a layout XML file if your own and see if it crashes?

```xml
            <android.support.design.widget.TextInputLayout
                android:id="@+id/password_layout"
                android:labelFor="@+id/password"
                android:layout_width="match_parent"
                android:layout_height="wrap_content"
                app:errorEnabled="true"
                tools:ignore="UnusedAttribute">

                <android.support.design.widget.TextInputEditText
                    android:id="@+id/password"
                    android:inputType="textPassword"
                    style="@style/FirebaseUI.TextInputEditText.EmailField" />

            </android.support.design.widget.TextInputLayout>
```

The result should be a password box.

**Comment by noxtrip** [comment]
> Ok I want to determine if this is actually an issue with FirebaseUI or just an issue with your app that is becoming visible through FirebaseUI. Could you try adding some views like this to a layout XML file if your own and see if it crashes?
> 
> ```
>             <android.support.design.widget.TextInputLayout
>                 android:id="@+id/password_layout"
>                 android:labelFor="@+id/password"
>                 android:layout_width="match_parent"
>                 android:layout_height="wrap_content"
>                 app:errorEnabled="true"
>                 tools:ignore="UnusedAttribute">
> 
>                 <android.support.design.widget.TextInputEditText
>                     android:id="@+id/password"
>                     android:inputType="textPassword"
>                     style="@style/FirebaseUI.TextInputEditText.EmailField" />
> 
>             </android.support.design.widget.TextInputLayout>
> ```
> 
> The result should be a password box.

Hi Sam,
thank you for your answer and suggestion. I have added simple new activity in my app with following layout:
```<?xml version="1.0" encoding="utf-8"?>
<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="match_parent"
    android:layout_height="match_parent"> 


<android.support.design.widget.TextInputLayout
    android:id="@+id/password_layout"
    android:labelFor="@+id/password"
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    app:errorEnabled="true"
    tools:ignore="UnusedAttribute">
    <android.support.design.widget.TextInputEditText
        android:id="@+id/password"
        android:inputType="textPassword"
        style="@style/FirebaseUI.TextInputEditText.EmailField" />
</android.support.design.widget.TextInputLayout>
</android.support.constraint.ConstraintLayout>```

and set android:theme:"@style/Theme.AppCompact.Light" in manifest for that activity.

I tried to call it with `.startActivity(intent) ` method and I got the same error as when using FirebaseUI email login. 

>  Caused by: android.view.InflateException: Binary XML file line #8: Binary XML file line #8: Error inflating class android.support.design.widget.TextInputLayout

Based on this [(similiar issue)](https://stackoverflow.com/questions/55170605/firebaseui-inflating-exception) i have added `app:passwordToggleDrawable="@null"` into the `android.support.design.widget.TextInputLayout` element and I was able to call that activity and display password field. I am really not sure where the error is than but maybe if I could add this property to FirebaseUI layout it could help? Do you by any chance have idea how to achieve this?

Thank you very much.

**Comment by samtstern** [comment]
Hmmm I don't think we actually want to add this change into FirebaseUI.  The visibility toggle is important in making the password field usable.

I *think* this probably has something to do with how your app is using vector drawables.  Unfortunately now that we've determined this is not a problem in FirebaseUI (since it occurs any time you try to use `TextInputEditLayout` in your app) I can't really dive into this and debug.  I hope you understand!

You may have better luck posting on StackOverflow.  I'd remove any mention of FirebaseUI and just show how your app crashes when you use the XML from your previous comment.

**Comment by noxtrip** [comment]
> Hmmm I don't think we actually want to add this change into FirebaseUI. The visibility toggle is important in making the password field usable.
> 
> I _think_ this probably has something to do with how your app is using vector drawables. Unfortunately now that we've determined this is not a problem in FirebaseUI (since it occurs any time you try to use `TextInputEditLayout` in your app) I can't really dive into this and debug. I hope you understand!
> 
> You may have better luck posting on StackOverflow. I'd remove any mention of FirebaseUI and just show how your app crashes when you use the XML from your previous comment.

Sure I understand. Thanks for help and time anyway.

**Comment by noxtrip** [comment]
Hi Sam,

just for others who might have similiar issue I was able to fix it and use also email sign in method with FirebaseUI. It is funny as actually your answer on other topic from 3 years ago helped me to solve it [link](https://github.com/firebase/FirebaseUI-Android/issues/427).

So for anyone out there having this issue what I did: was in my application I have created res/layout folder. I have created copies of 4 layouts that FirebaseUI 4.3.1 uses when working with email signin (this is version dependent so they can differ for you). In my case these were:

fui_forgot_password_layout.xml, fui_check_email_layout.xml, fui_register_email_layout.xml, fui_register_email_layout.xml

In each of these files wherever there was` <android.support.design.widget.TextInputLayout>` used I have added this property `app:passwordToggleDrawable="@null"` and for password fields I have also removed `app:passwordToggleEnabled="true"`. This is example of TextInputLayout after changes:
```<android.support.design.widget.TextInputLayout
                android:id="@+id/password_layout"
                style="@style/FirebaseUI.TextInputLayout.PasswordField"
                android:layout_width="match_parent"
                android:layout_height="wrap_content"
                android:layout_marginTop="@dimen/fui_field_padding_vert"
                app:layout_constraintTop_toBottomOf="@+id/welcome_back_password_body"
				app:passwordToggleDrawable="@null">

                <android.support.design.widget.TextInputEditText
                    android:id="@+id/password"
                    style="@style/FirebaseUI.TextInputEditText.PasswordField" />

            </android.support.design.widget.TextInputLayout>
```
I know I have lost possibilty to show password but I am willing to omit the functionality for sake of using FirebaseUI as it is as I like it.

Cheers
